plugins/source/azure/resources/services/costmanagement/views.go

```diff
@@ -20,20 +20,18 @@ func Views() *schema.Table {
 			transformers.WithNameTransformer(client.ETagNameTransformer),
 			transformers.WithPrimaryKeys("ID"),
 		),
-		Columns: schema.ColumnList{client.SubscriptionID},
-		Relations: []*schema.Table{
-			view_queries(),
-		},
+		Columns:   schema.ColumnList{client.SubscriptionIDPK},
+		Relations: schema.Tables{view_queries()},
 	}
 }
 
-func fetchViews(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
+func fetchViews(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- any) error {
 	cl := meta.(*client.Client)
 	svc, err := armcostmanagement.NewViewsClient(cl.Creds, cl.Options)
 	if err != nil {
 		return err
 	}
-	pager := svc.NewListByScopePager("subscriptions/"+cl.SubscriptionId, nil)
+	pager := svc.NewListPager(nil)
```

Comment:
If no subscription then you need to update multiplexer to suit.

Reply:
Also doing this fails to get any of the 'shared views' as it was verified by api call in the original issue. I think shared views use case would be more important in large teams.

Reply:
Yeah, I just implemented this to be discussed during PR review